pb-rs/src/types.rs

```diff
@@ -512,67 +545,103 @@ impl Field {
                     self.typ.get_size("")
                 )?;
             }
+            Frequency::Optional => writeln!(
+                w,
+                "if self.{} == {} {{ 0 }} else {{ {} + {} }}",
+                self.name,
+                self.default
+                    .iter()
+                    .next()
+                    .map(|s| s.as_str())
+                    .unwrap_or(self.typ.regular_default(desc).unwrap_or("None")),
```

Comment:
```
self.default.as_ref().map_or_else(|| self.typ.regular_default(desc).unwrap_or("None"), s.as_str())
```

Reply:
Did you mean (added a `|s|` to `s.as_str()`):

```rust
self.default.as_ref().map_or_else(|| self.typ.regular_default(desc).unwrap_or("None"), |s| s.as_str())
```

Reply:
yep